source/simulator/assembly.cc

```diff
@@ -870,25 +870,6 @@ namespace aspect
         scratch.finite_element_values[solution_field].get_function_laplacians (old_old_solution,
                                                                                scratch.old_old_field_laplacians);
 
-        compute_material_model_input_values (current_linearization_point,
-                                             scratch.finite_element_values,
-                                             cell,
-                                             true,
-                                             scratch.material_model_inputs);
-        material_model->evaluate(scratch.material_model_inputs,scratch.material_model_outputs);
-        if (advection_field.is_temperature()==true)
-          {
-            MaterialModel::MaterialAveraging::average (parameters.material_averaging,
-                                                       cell,
-                                                       scratch.finite_element_values.get_quadrature(),
-                                                       scratch.finite_element_values.get_mapping(),
-                                                       scratch.material_model_outputs);
-            HeatingModel::HeatingModelOutputs heating_model_outputs(n_q_points, parameters.n_compositional_fields);
-            heating_model_manager.evaluate(scratch.material_model_inputs,
-                                           scratch.material_model_outputs,
-                                           heating_model_outputs);
```

Comment:
how come we don't need this? Doesn't the computation of the residual of the temperature equation depend on what heating terms we have?